tools/file_packager.py

```diff
@@ -291,6 +291,14 @@ def main():
   }
   Module.expectedDataFileDownloads++;
   (function() {
+   // When running as a pthread, FS operations are proxied to the main thread, so we don't need to
+   // fetch the .data bundle on the worker
+   if (Module['ENVIRONMENT_IS_PTHREAD']) {
+    // This piece of code handles the case when building with assertions enabled
+    // See https://github.com/emscripten-core/emscripten/blob/4d864df0a57024d6/tools/js_manipulation.py#L20
+    if (!Module['preRun']) Module['preRun'] = [];
```

Comment:
This doesn't seem like it should be necessary.. we should probably fix js_manipulation.py to be smarter... I'll honestly don't know what it that is for but I will check it out.

Reply:
It looks like we can modify `js_manipulation.py` such that it also wraps its logic in `if (Module['ENVIRONMENT_IS_PTHREAD']) {`

Reply:
Thanks, let me know if there is a better way to fix this. 
It looks like with assertions enabled, we are trying to enforce that the .data file has been processed (https://github.com/emscripten-core/emscripten/blob/4d864df0a57024d6/tools/js_manipulation.py#L22) and the presence of `Module['preRun']` is used to determine that.